bin/maybe-skip-tests.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Paths to ignore.
+ignored_paths=(
+  .editorconfig
+  .gitattributes
+  .gitignore
+  CODEOWNERS
+  CONTRIBUTING.md
+  LICENSE
+  phpcs.xml.dist
+  phpunit.xml.dist
+  README.md
+  readme.txt
+  .circleci/config.yml
+  bin/maybe-skip-tests.sh
+  bin/helpers.sh
+  bin/install-local-tests.sh
+  bin/install-wp-tests.sh
+  bin/phpunit-test.sh
+  .wordpress-org/*
+  .github/*
+  tests/phpunit/*
+)
+
+# Fetch list of changed files from the last commit
+changed_files=$(git diff-tree --no-commit-id --name-only -r HEAD)
+should_run_tests=true
+
+for file in $changed_files; do
+	is_ignored=false
+	for ignore in "${ignored_paths[@]}"; do
+		if [ "$file" = "$ignore" ]; then
+			is_ignored=true
+			break
+		fi
+	done
+	if [ "$is_ignored" = false ]; then
+		should_run_tests=true
+		echo "Running tests because $file was changed."
+		break
+	else
+		should_run_tests=false
+		echo "Skipping $file..."
+	fi
+done
```

Comment:
Nesting loops is fun. I would write helper functions to clean this up (also not re-defining should runt ests because it was set above).

```suggestion

is_ignored_file(){
    for ignore in "${ignored_paths[@]}"; do
		if [[ "${1:-}" == "$ignore" ]]; then
			return 0
		fi
	done
	return 1
}

for file in $changed_files; do
    if ! is_ignored_file "$file"; then
		echo "Running tests because $file was changed."
		break
    fi
    should_run_tests=false
	echo "Skipping $file..."
done
```